### DISCUSSION TITLE
encode_special_tokens=True 导致防注入失效

### DISCUSSION THREAD

**Discussion by Jingsong-Yan** [zai-org/ChatGLM3#415] [Q&A] [answered]
为了适配fschat，tokenization_chatglm.py加了encode_special_tokens=True后，防注入就失效了，多轮对话可以拼接成一个句子，然后在转化为id，<|user|>就是一个token。而防注入希望，当用户输入<|user|>，会被tokenize 成多个token。
 https://huggingface.co/THUDM/chatglm3-6b/blob/e46a14881eae613281abbd266ee918e93a56018f/tokenization_chatglm.py#L35

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
mark
